src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const clearProp = 'clear';
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    if (action.type === add) {
```

Comment:
This here is a perfect case for a [switch statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) rather than a bunch of ifs